packages/smooth_app/lib/pages/product/new_product_page.dart

```diff
@@ -0,0 +1,274 @@
+import 'dart:ui';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/model/Attribute.dart';
+import 'package:openfoodfacts/model/Product.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/cards/data_cards/image_upload_card.dart';
+import 'package:smooth_app/cards/data_cards/score_attribute_card.dart';
+import 'package:smooth_app/cards/expandables/attribute_list_expandable.dart';
+import 'package:smooth_app/data_models/product_preferences.dart';
+import 'package:smooth_app/database/dao_product_extra.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/helpers/attributes_card_helper.dart';
+import 'package:smooth_app/helpers/launch_url_helper.dart';
+import 'package:smooth_app/pages/product/common/product_dialog_helper.dart';
+import 'package:smooth_ui_library/widgets/smooth_card.dart';
+
+class NewProductPage extends StatefulWidget {
+  const NewProductPage({
+    required this.product,
+  });
+
+  final Product product;
+
+  @override
+  State<NewProductPage> createState() => _ProductPageState();
+}
+
+class _ProductPageState extends State<NewProductPage> {
+  late Product _product;
+  bool _first = true;
+
+  @override
+  void initState() {
+    super.initState();
+    _updateHistory(context);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final LocalDatabase localDatabase = context.watch<LocalDatabase>();
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    if (_first) {
+      _first = false;
+      _product = widget.product;
+    }
+    const Color backgroundColor = Color.fromARGB(255, 234, 244, 234);
```

Comment:
Magic number: better from theme or constants.

Reply:
this specific color is not in theme, I got it from the mocks. But maybe it's better as a constant in this class, but it might also make sense to put it in one of the shared classes and access from there, in case it's adopted as a background color in another page, what would you recommend ? I'm unfamiliar with the best practices in dart, open to suggestions.

Reply:
From my 18 month old flutter training, I would recommend a constant, something like `const Color kWhateverBackgroundColor = Color.fromARGB(255, 234, 244, 234);` in a specific constant file. Maybe we already have one in this project.

Reply:
Alright, I agree, sharing it in a class is better. I've put it in SmoothTheme, as I saw usage of other color themes declared there from other parts of the app, but let me know if you think it should be kept in a file separate from "Theme" since it's not strictly part of any theme. In that case I would create a new const file a new shared/ package or so.